nextcloud/server
# Read and write time zone ID when updating CalDAV availability

Tiny bug/limitation of #27466

Now the timezone id is read from the previous value but also encoded in the newly saved icalendar value. If no time zone was set before it will still fall back to the auto detected one.

![Bildschirmfoto von 2021-10-19 13-43-56](https://user-images.githubusercontent.com/1374172/137905369-5ffd50fa-9396-462e-a583-9247e90410de.png)

## How to test
* Open `/settings/user/groupware`
* Set the timezone to something other than the auto-detected value
* Save the availability
* Reload the page

On master: you see the auto detected time zone again
On this branch: time zone is the one you previously stored

Without this fix reloading the page and saving the availability will shift your times by the drift of the time zone.

I tested this against an availability rule created by Apple Calendar. ~~The only diff to their format is that we don't generate the DST vs standard time recurrence rules within the timezone component but I guess we would need some data for this that we currently don't have.~~ done with the help of https://www.npmjs.com/package/icalzone
